src/toolchain.rs

```diff
@@ -685,12 +686,15 @@ impl<'a> DistributableToolchain<'a> {
         allow_downgrade: bool,
         components: &[&str],
         targets: &[&str],
+        profile: Option<Profile>,
     ) -> Result<UpdateStatus> {
         let update_hash = self.update_hash()?;
         let old_date = self.get_manifest().ok().and_then(|m| m.map(|m| m.date));
         InstallMethod::Dist {
             desc: &self.desc()?,
-            profile: self.0.cfg.get_profile()?,
+            profile: profile
+                .map(|p| Ok(p))
+                .unwrap_or_else(|| self.0.cfg.get_profile())?,
```

Comment:
```suggestion
            profile: profile.map_or_else(Ok, || self.0.cfg.get_profile())?,
```
or if you don't like `map_or_else` :P just changing the closure to `Ok` directly.

Reply:
I keep forgetting that variant constructors are functions, I also was unaware of `.map_or_else()` which I now need to add to my brain.